services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/helpers/Wrappers.groovy

```diff
@@ -0,0 +1,93 @@
+package org.openkilda.functionaltests.helpers
+
+import groovy.util.logging.Slf4j
+import org.junit.runners.model.MultipleFailureException
+
+@Slf4j
+class Wrappers {
+
+    /**
+     * Retries some operation for a given number of times. Retry is considered failed if any throwable is thrown.
+     * Throws MultipleFailureException if closure fails to execute in given amount of tries.
+     *
+     * @param times number of retries
+     * @param retryInterval pause between retries (in seconds)
+     * @param handler exception handler (receives exception on each failure)
+     * @param body operation to wrap around
+     * @return result of the body execution
+     */
+    def static retry(int times = 5, double retryInterval = 2, Closure handler = { e -> log.debug(e) }, Closure body) {
+        int retries = 0
+
+        List<Throwable> ex = []
+        while (retries++ < times) {
+            try {
+                return body.call()
+            } catch (Throwable t) {
+                ex.add(t)
+                handler.call(t)
+                sleep(retryInterval * 1000 as long)
+            }
+        }
+        throw new MultipleFailureException(ex)
+    }
+
+    /**
+     * Wait for closure to return a result. Returns false if success condition was not met.
+     *
+     * @param timeout time to wait in seconds
+     * @param retryInterval pause between retries (in seconds)
+     * @param closure : A closure that returns following:
+     *          - True if check is successful
+     *          - False or null if check is unsuccessful
+     * @return True if wait was successful, false otherwise
```

Comment:
Are you planning to add something in the @throws section?

Reply:
this method never throws

Reply:
Hah, yes. You're catching all throwables. Agree. We don't need this section here.